pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/mapper/HadoopSegmentCreationMapReduceJob.java

```diff
@@ -83,17 +81,27 @@ public void setup(Context context) throws IOException, InterruptedException {
       LOGGER.info("*********************************************************************");
       _properties = context.getConfiguration();
 
-      _outputPath = _properties.get("path.to.output");
-      _tableName = _properties.get("segment.table.name");
+      _outputPath = _properties.get(JobConfigConstants.PATH_TO_OUTPUT);
+      _tableName = _properties.get(JobConfigConstants.SEGMENT_TABLE_NAME);
       _postfix = _properties.get("segment.name.postfix", null);
 
       if (_outputPath == null || _tableName == null) {
         throw new RuntimeException(
             "Missing configs: " +
                 "\n\toutputPath: " +
-                _properties.get("path.to.output") +
+                _properties.get(JobConfigConstants.PATH_TO_OUTPUT) +
                 "\n\ttableName: " +
-                _properties.get("segment.table.name"));
+                _properties.get(JobConfigConstants.SEGMENT_TABLE_NAME));
+      }
+      String tableConfigString = _properties.get(JobConfigConstants.TABLE_CONFIG);
+      if (tableConfigString != null) {
+        try {
+          _tableConfig = TableConfig.init(tableConfigString);
+        } catch (JSONException e) {
+          // Though we get table config directly from the controller of hosts and port of push location are set,
+          // it is possible for the user to pass in a table config as a parameter
+          LOGGER.error("Table config {} isn't passed in correctly", tableConfigString);
```

Comment:
It'd be good to print the conclusion and then show the incorrect config, like:
String.format("Exception when parsing table config: %s", tableConfigString);